elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/dialects/SQLDialectFactory.java

```diff
@@ -13,19 +13,36 @@
  * A class with static methods to create an instance of all Dialects.
  */
 public class SQLDialectFactory {
+
+    private static final SQLDialect H2_DIALECT = new H2Dialect();
+    private static final SQLDialect HIVE_DIALECT = new HiveDialect();
+    private static final SQLDialect PRESTO_DIALECT = new PrestoDialect();
+
     public static SQLDialect getDefaultDialect() {
-        return new H2Dialect();
+        return getH2Dialect();
     }
 
     public static SQLDialect getH2Dialect() {
-        return new H2Dialect();
+        return H2_DIALECT;
     }
 
     public static SQLDialect getHiveDialect() {
-        return new HiveDialect();
+        return HIVE_DIALECT;
     }
 
     public static SQLDialect getPrestoDialect() {
-        return new PrestoDialect();
+        return PRESTO_DIALECT;
+    }
+
+    public static SQLDialect getDialect(String type) {
```

Comment:
Instead of instantiation via string name, we could pass the class name here and use reflection to load the dialect like a plugin.

Reply:
ok.